src/main/java/bio/terra/cli/service/utils/HttpUtils.java

```diff
@@ -108,4 +118,70 @@ public static HttpResponse sendHttpRequest(
     // return a POJO that includes both the response body and status code
     return new HttpResponse(responseBody.toString(), statusCode);
   }
+
+  /**
+   * Helper method to do retries. Uses {@link #DEFAULT_MAXIMUM_RETRIES} for maximum number of
+   * retries and {@link #DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY} for the time to sleep between
+   * retries.
+   *
+   * @param makeRequest function to perform the request
+   * @param isRetryable function to test whether the exception is retryable or not
+   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
+   * @return the Http response
+   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
+   *     number of retries was exhausted
+   */
+  public static <T> T callWithRetries(
+      HttpRequestOperator<T> makeRequest, Predicate<Exception> isRetryable) throws Exception {
+    return callWithRetries(
+        DEFAULT_MAXIMUM_RETRIES, DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY, makeRequest, isRetryable);
+  }
+  /**
+   * Helper method to do retries.
+   *
+   * @param maxRetries maximum number of times to retry
+   * @param sleepDurationMS number of milliseconds to sleep between tries
+   * @param makeRequest function to perform the request
+   * @param isRetryable function to test whether the exception is retryable or not
+   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
+   * @return the Http response
+   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
+   *     number of retries was exhausted
+   */
+  public static <T> T callWithRetries(
+      int maxRetries,
+      int sleepDurationMS,
+      HttpRequestOperator<T> makeRequest,
+      Predicate<Exception> isRetryable)
+      throws Exception {
+    int numTries = 0;
+    do {
+      numTries++;
+      try {
+        logger.info("Http request attempt #{}", numTries);
+        return makeRequest.makeRequest();
+      } catch (Exception ex) {
+        // if the exception is not retryable, then quit polling
+        if (!isRetryable.test(ex)) {
+          throw ex;
+        }
+        logger.info("Caught retryable exception: {}", ex);
+      }
+
+      // sleep before retrying
+      Thread.sleep(sleepDurationMS);
+    } while (numTries <= maxRetries);
+
+    throw new RuntimeException("Http request with retries timed out after " + numTries + " tries.");
+  }
+
+  /**
+   * Function interface for making a retryable Http request. This interface is explicitly defined so
+   * that it can throw an exception (i.e. Supplier does not have this method annotation).
+   *
+   * @param <T> type of the Http response (i.e. return type of the makeRequest method)
+   */
+  public interface HttpRequestOperator<T> {
+    T makeRequest() throws Exception;
```

Comment:
we can template this on the exception being thrown as well that way we don't have to throw a generic  `Exception`. `HttpRequestOperator<T, E> { T makeRequest() throws E; }` But it doesn't look like any of our functions throw checked exceptions yet, so we do not need an exception throwing functional interface at this time. 
When we want to do retries on functions that both throw checked exceptions and don't, then we'll also want a checked exception version.

Reply:
Adding an additional template for the exception type seems good -- I'll give that a try.